etrago/cluster/snapshot.py

```diff
@@ -230,218 +255,312 @@ def tsam_cluster(timeseries_df,
-
-                        if new_date.isin(timeseries.index.get_level_values('dates')):
-                            timeseries['dates'] = timeseries.index.get_level_values('dates')
-                            timeseries['SegmentNo'] = timeseries.index.get_level_values('SegmentNo')
-                            timeseries['SegmentDuration'] = timeseries['SegmentDuration_Extreme']
-                            timeseries.drop('SegmentDuration_Extreme', axis=1, inplace=True)
-                            timeseries.set_index(['dates', 'SegmentNo', 'SegmentDuration'],inplace=True)
-                            break
-                        else:
-                            new_row = timeseries_df.iloc[new_row].copy()
-                            new_row.drop('row_no', inplace=True)
-                            new_row['SegmentNo'] = len(timeseries)
-                            new_row['SegmentDuration'] = old_row['SegmentDuration_Extreme'][0] - delta_t - 1
-                            new_row['dates'] = new_row.name
-                            new_row = pd.DataFrame(new_row).transpose()
-                            new_row.set_index(['dates', 'SegmentNo', 'SegmentDuration'],inplace=True)
-                            for col in new_row.columns:
-                                new_row[col][0] = old_row[col][0]
-
-                            timeseries['dates'] = timeseries.index.get_level_values('dates')
-                            timeseries['SegmentNo'] = timeseries.index.get_level_values('SegmentNo')
-                            timeseries['SegmentDuration'] = timeseries['SegmentDuration_Extreme']
-                            timeseries.drop('SegmentDuration_Extreme', axis=1, inplace=True)
-                            timeseries.set_index(['dates', 'SegmentNo', 'SegmentDuration'],inplace=True)
-                            timeseries = timeseries.append(new_row)
-                            timeseries = timeseries.sort_values(by='dates')
-                            break
-
-            elif extremePeriodMethod == 'replace_cluster_center':
-
-                # replace segment in which the extreme timestep was added
-                i=-1
-                for date in timeseries.index.get_level_values('dates'):
-                    if date < maxi:
-                        i = i+1
-                    else:
-                        if i ==-1:
-                            i=0
-                        max_val['SegmentDuration'] = timeseries.index.get_level_values('SegmentDuration')[i]
-                        max_val.set_index(['dates', 'SegmentNo', 'SegmentDuration'],inplace=True)
-                        timeseries.drop(timeseries.index[i], inplace=True)
-                        timeseries = timeseries.append(max_val)
-                        timeseries = timeseries.sort_values(by='dates')
+def segmentation_extreme_periods(
+    timeseries_df, timeseries, extremePeriodMethod
+):
```

Comment:
There are no docstrings yet, but you will add them later on on the documentation branch, right?
If yes, it's fine by now.

Reply:
Yes, they are added in the documentation branch: https://github.com/openego/eTraGo/tree/documentation-release0.9.0